.github/drafts/16521/Code-scanner-Draft.md

```diff
@@ -0,0 +1,122 @@
+
+[Text Video](https://youtu.be/o80-IKcXAVQ)
+
+
+### Value
+
+Value is a binding property, it fetches the response from the scanned code. To bind the value of the code scanner widget to another widget, open the property pane of it, and add the code snippet given below:
+
+```
+{{<your_widget>.value}}
+```
+
+
+For example, let's take a Code Scanner widget(CodeScanner1) and bind its value to a text widget. Drag a text widget onto the canvas and add the following code:
+
+```
+{{CodeScanner1.value}}
+```
+
+.https://youtu.be/zfkpIzaiTX0
+
+
+# Events
+
+They are a set of actions that you can perform on the widget. The following table lists the actions:
+
+| Event          |  Description                                     |                                            
+|----------------|--------------------------------------------------|
+| onCodeDetected | Triggers an action when a valid code is detected. See a list of [supported actions](https://docs.appsmith.com/reference/appsmith-framework/widget-actions).|
+
+### onCodeDetected 
+
+When a valid barcode or QR code is detected, this event triggers an action. 
+The Appsmith framework allows triggering actions for widget events and inside JS Objects. Let’s take an example to understand how the onCodeDetected event works.
+
+  
+In this example, we will use the [Modal widget](https://docs.appsmith.com/reference/widgets/modal) to display the code response.
+
+* First, set the onCodeDetected event to open a New Modal, and choose to Create New.
+* This will open up a new modal; let's drag the [text widget](https://docs.appsmith.com/reference/widgets/text) into the modal.
+* In the property section, add the following code:
+  
+```
+{{CodeScanner1.value}}
+```
+
+With this, whenever a valid code is detected, a modal window displaying the code response will show up. The video below demonstrates how to achieve that.
+  
+https://youtu.be/_2p1bMbdk6U
+
+
+# Styles
+  Style properties allow you to change the look and feel of the widget. 
+  | Style          | Description                                            |
+|----------------|--------------------------------------------------------|
+| Icon           | Sets an icon to be included in the input field.        |
+| Position       | Sets the label position of the widget.                 |
+| Placement      | Sets the label alignment of the widget.                |
+| Button Color   | Allows you to set color for the button.                |
+|  Border Radius | Allows you to define curved corners.                   |
+| Box Shadow     | Allows you to choose from the available shadow styles. |
+
+Code Scanner is a beneficial widget for your business. It is efficient, cost-effective, and time-effective.
```

Comment:
IMO, this conclusion para looks a bit unnecessary for documentation. cc: @jnikhila

Reply:
+1

Reply:
+1